### PR TITLE
port tests in test_functional_tensor to pytest

### PR DESCRIPTION
**Pull request by sahilg06**
Refactored the following tests to pytest in the file test/test_functional_tensor.py mentioned in issue #3956

Group A

- test_assert_image_tensor
- test_vflip
- test_hflip
- test_crop 

Group E

- test_gaussian_blur

### CODE REVIEW

File: test/test_functional_tensor.py [hunk 1]
```diff
@@ -1141,5 +1003,148 @@ def test_adjust_gamma(device, dtype, config):
     )
 
 
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('func, args', [(F_t._get_image_size, ()), (F_t.vflip, ()),
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:1007]: It's no big deal but I find this formatting to be a bit cleaner, and easier to read:
```py
@pytest.mark.parametrize('func, args', [
    (F_t._get_image_size, ()),
    (F_t.vflip, ()),
    (F_t.hflip, ()),
    (F_t.crop, (1, 2, 4, 5)),
    (F_t.adjust_brightness, (0., )),
    (F_t.adjust_contrast, (1., )),
    ...
])
```

File: test/test_functional_tensor.py [hunk 2]
```diff
@@ -1141,5 +1003,148 @@ def test_adjust_gamma(device, dtype, config):
...
+    # scriptable function test
+    hflipped_img_script = script_hflip(img_tensor)
+    assert_equal(hflipped_img, hflipped_img_script)
+
+    batch_tensors = _create_data_batch(16, 18, num_samples=4, device=device)
+    _test_fn_on_batch(batch_tensors, F.hflip)
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('top,left,height,width', [(1, 2, 4, 5),   # crop inside top-left corner
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:1062]: We could do something similar here

File: test/test_functional_tensor.py [hunk 3]
```diff
@@ -1141,5 +1003,148 @@ def test_adjust_gamma(device, dtype, config):
...
+    p = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'assets', 'gaussian_blur_opencv_results.pt')
+    true_cv2_results = torch.load(p)
+
+    if image_size == 'small':
+
+        tensor = torch.from_numpy(
+            np.arange(3 * 10 * 12, dtype="uint8").reshape((10, 12, 3))
+        ).permute(2, 0, 1).to(device)
+
+    else:
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:1116]: no need to skip lines here:

```suggestion
    if image_size == 'small':
        tensor = torch.from_numpy(
            np.arange(3 * 10 * 12, dtype="uint8").reshape((10, 12, 3))
        ).permute(2, 0, 1).to(device)
    else:
```

**Review by NicolasHug — APPROVED** [review]
Thanks @sahilg06 ! This looks great, I only have some minor cosmetics comments, but LGTM when green!

**Comment by sahilg06** [comment]
> Thanks @sahilg06 ! This looks great, I only have some minor cosmetics comments, but LGTM when green!

Thanks a lot @NicolasHug
 So should I do the suggested cosmetic changes or its fine to merge?

**Comment by NicolasHug** [comment]
> So should I do the suggested cosmetic changes

Yes ;), I tend to approve early when the requested changes are easy / obvious. I sometimes make comments that don't really need to be addressed and for those I usually say they're Nits. The  comments here were bordeline nits.

Thanks a lot for the PR!

**Pull request merged**